### PR TITLE
feat(Spanner): Add directed read samples

### PR DESCRIPTION
**Pull request by ajupazhamayil**
This is dependent on https://github.com/googleapis/google-cloud-php/pull/6078
Once the dependent PR merged, need to update the `composer.json` file to match the `google-cloud-php` latest version.

### CODE REVIEW

File: spanner/src/directed_read.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright 2023 Google LLC.
```
Comment (vishwarajanand) [inline spanner/src/directed_read.php:3]: 😼 
```suggestion
 * Copyright 2024 Google LLC.
```

File: spanner/src/directed_read.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/main/spanner/README.md
+ */
+
+namespace Google\Cloud\Samples\Spanner;
+
+// [START spanner_directed_read]
```
Comment (vishwarajanand) [inline spanner/src/directed_read.php:26]: Are you sure this is the correct tag? Only C++ seems to have implemented this!

**Review by vishwarajanand — APPROVED** [review]
Lgtm, approving to unblock.
Pls check the `region_tag` and run tests once before merge.

File: spanner/src/directed_read.php (hunk 2 above)
Comment (ajupazhamayil) [reply spanner/src/directed_read.php:26]: Yes! Its mentioned in the shared design doc.

**Pull request merged**